### DISCUSSION TITLE
different raw message retention days

### DISCUSSION THREAD

**Discussion by Phillipip** [postalserver/postal#3192] [Feature suggestions]
It would be great if you could define different retention periods. I would like to delete messages that were successfully delivered, but I would like to keep messages that were rejected a little longer so that the error can be fixed (which can take more than a day) and the mail can then be sent again.

**Comment by willpower232** [comment]
Postal isn't really intended to keep messages very long, you would probably do better to implement webhooks so you can get more directly notified of the status of all your messages. This also means you can keep extra context for the messages for your future reference.
